extensions/notebook/src/book/bookTrustManager.ts

```diff
@@ -0,0 +1,161 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as constants from './../common/constants';
+import { BookTreeItem } from './bookTreeItem';
+import { BookModel } from './bookModel';
+
+export interface IBookTrustManager {
+	isNotebookTrustedByDefault(notebookUri: string): boolean;
+	setBookAsTrusted(bookRootPath: string): boolean;
+	setBookAsUntrusted(bookRootPath: string): boolean;
+}
+
+export interface IBookTrustManagerWorkspaceDetails {
+	workspaceFolders: vscode.WorkspaceFolder[];
+	getConfiguration(section?: string | undefined): vscode.WorkspaceConfiguration;
+}
+
+enum TrustBookOperation {
+	Add,
+	Remove
+}
+
+export class BookTrustManager implements IBookTrustManager {
+
+	private static notebookConfiguration: string = constants.notebookConfigKey;
+	private static notebookTrustedBooksConfiguration: string = constants.trustedBooksConfigKey;
+	private trustedLocalBooks: string[] = [];
+
+	constructor(private books: BookModel[], private workspaceDetails?: IBookTrustManagerWorkspaceDetails) {
+		if (!workspaceDetails) {
+			this.workspaceDetails = {
+				get getConfiguration() {
+					return vscode.workspace.getConfiguration;
+				},
+				get workspaceFolders() {
+					return [].concat(vscode.workspace.workspaceFolders || []);
+				}
+			};
+		}
+	}
+
+	isNotebookTrustedByDefault(notebookUri: string): boolean {
+		let allBooks: BookTreeItem[] = this.getAllBooks();
+		let trustableBookPaths = this.getTrustableBookPaths();
+		let trustableBooks: BookTreeItem[] = allBooks
+			.filter(bookItem => trustableBookPaths.some(trustableBookPath => trustableBookPath === path.join(bookItem.book.root, path.sep)));
+		let isNotebookTrusted = this.isNotebookPartOfBooks(notebookUri, trustableBooks);
+
+		return isNotebookTrusted;
+	}
+
+	isNotebookPartOfBooks(notebookUri: string, books: BookTreeItem[]) {
```

Comment:
>isNotebookPartOfBooks [](start = 1, length = 21)

When you do have access to BookModel, it has get all notebooks method, which gives you the list in the toc. Why not check directly if the path exists on the list?

Reply:
Moved logic out to `BookTreeItem`. See above comment. #resolved